snuba/datasets/transactions_processor.py

```diff
@@ -39,6 +41,18 @@ class TransactionsMessageProcessor(MessageProcessor):
         "sentry:dist",
     }
 
+    def __merge_nested_field(self, keys: Sequence[str], values: Sequence[str]) -> str:
+        def escape_field(field: str) -> str:
+            """
+            We have ':' in our tag names. Also we may have '|'. This escapes : and \ so
+            that we can always rebuild the tags from the map. When looking for tags with LIKE
+            there should be no issue. But there may be other cases.
+            """
+            return field.translate(ESCAPE_TRANSLATION)
+
+        pairs = [f"{escape_field(k)}:{escape_field(v)}" for k, v in zip(keys, values)]
```

Comment:
~This should probably be a generator rather than a list comprehension.~

Nevermind — I figured I should check to make sure this would actually help, turns out it probably makes no difference: https://stackoverflow.com/questions/34822676/joining-strings-generator-or-list-comprehension